File: articles/cascading-risk-and-systemic-decision-failure/julia/threshold_failure_model.jl
# threshold_failure_model.jl

function threshold_failure(stress, neighbor_failure_load, buffer, threshold)
    effective_stress = stress + neighbor_failure_load + max(0.0, 0.40 - buffer)
    return effective_stress >= threshold
end

println("Threshold failure? ", threshold_failure(0.52, 0.18, 0.31, 0.66))

File: articles/cascading-risk-and-systemic-decision-failure/python/threshold_failure_model.py
#!/usr/bin/env python3
"""Threshold failure helper."""

from __future__ import annotations


def threshold_failure(stress: float, neighbor_failure_load: float, buffer: float, threshold: float) -> bool:
    effective_stress = stress + neighbor_failure_load + max(0.0, 0.40 - buffer)
    return effective_stress >= threshold


if __name__ == "__main__":
    print(threshold_failure(0.52, 0.18, 0.31, 0.66))
